lib/mutant/mutation.rb

```diff
@@ -40,11 +40,11 @@ def original_source
 
     # Test if mutation is killed by test reports
     #
-    # @param [Array<Report::Test>] test_reports
+    # @param [Result::Test] test_result
     #
     # @return [Boolean]
-    def self.success?(test_result)
-      self::TEST_PASS_SUCCESS.equal?(test_result.passed)
+    def success?(test_result)
```

Comment:
> It is de-facto using instance state self.class:: for looking up a constant relative to the instances class.

Well, sort of. It doesn't require instance state. You could make due with class state.

Also I wasn't factoring in the constant at all. That isn't relevant imho. My primary consideration is if actual instance state is used or not. If it's not, then the method should be a singleton method.

My point in flagging this is that when you could rewrite the method as the following and still have it pass, then maybe it shouldn't be an instance method:

```ruby
def success?(test_result)
  self.class.success?(test_result)
end

def self.success?(test_result)
  self::TEST_PASS_SUCCESS.equal?(test_result.passed)
end
```

You could also just use `delegate :success?, to: self`if you had a delegation helper.